ndinterp/src/grid.rs

```diff
@@ -20,24 +38,35 @@ pub mod cubic;
 ///     A d-dimensional vector of 1-dimensional sorted vectors for the input points
 ///     A d-dimensional array for the grid values of
 #[derive(Debug)]
-pub struct Grid<D: Dimension> {
+pub struct Grid<const D: usize>
+where
+    DimensionHelper<D>: ToDimension,
+{
     /// Arrays with the input vectors (x_i)
     pub xgrid: Vec<Vec<f64>>,
 
     /// Output points
-    pub values: Array<f64, D>,
+    pub values: Array<f64, <DimensionHelper<D> as ToDimension>::Dim>,
 }
 
 /// A grid slice is always 1-Dimensional
+/// and it is made of the x and y values such that f(x) = y
 #[derive(Debug)]
-pub struct GridSlice<'a> {
+pub(crate) struct GridSlice<'a> {
     /// A reference to one of the input vectors of the grid
     pub x: &'a Vec<f64>,
     /// A view of the slice of values corresponding to x
     pub y: ArrayView1<'a, f64>,
 }
 
-impl<'a> GridSlice<'a> {
+pub(crate) trait Derivatives<'a> {
```

Comment:
At a second thought: what's the motivation for making a `trait`, and not just implementing the methods?

Reply:
It looked more organized like this. "All derivatives are public for other members of the crate but only for them"

Reply:
You can have multiple `impl` blocks even without using traits, if that's the goal.

I also think it's nice to organize things, but I realized that it is not likely that anyone else will implement this trait, other than `GridSlice` itself (since you need a one dimensional set to derive over it).

So I have nothing to object about the organization, but if they are not going to be used twice, you could slightly simplify and save the prototypes.

Reply:
> You can have multiple impl blocks even without using traits, if that's the goal.

No, I mean the `pub(crate)` qualifiers. Because I thought that without the trait I had to qualify every function by itself. But I just discovered I was wrong, the `pub(crate)` in the struct itself is propagated.